[Ong Cheng Seong] iP

chengseong opened the pull request:
**DukePro**

> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember the things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER FAST_ to use

All you need to do is,

1. download it form [here](https://nus-cs2103-ay2122s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon) 
- [ ] Reminders (coming soon)

------------

If you Java programmer, you can use it to practice Java too. Here's the ``main`` method :

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

chengseong committed:
Level 1

chengseong committed:
Level 2

chengseong committed:
Level 3

chengseong committed:
Level 4

chengseong committed:
A-TextUiTesting

chengseong committed:
Level 5

chengseong committed:
Level 6

chengseong committed:
Level 7

chengseong committed:
Level 8

chengseong committed:
Merge branch 'branch-Level-8'

chengseong committed:
Settled merge conflicts and debugging

chengseong committed:
Created all new classes

chengseong committed:
Packages

chengseong committed:
Packages

chengseong committed:
Changed .gitignore to include packaged classes

chengseong committed:
JUnits

chengseong committed:
removed .class files

chengseong committed:
JavaDocs added and used switch for Parser instead of if else statements

chengseong committed:
minor commit to try merging

chengseong committed:
Merge branch 'branch-Level-7'

chengseong committed:
minor changes

chengseong committed:
Minor Changes

chengseong committed:
Added JavaDocs to Duke class

chengseong committed:
Level 9

chengseong committed:
Merge branch 'branch-A-JavaDoc'

chengseong committed:
Merged with branch branch-A-JavaDoc

chengseong committed:
Merge branch 'branch-A-CodingStandard'

chengseong committed:
Removed idea folder

chengseong committed:
Removed idea folder

chengseong committed:
Changed .gitignore

deetoomoo commented on src/main/java/AddCommand.java:
```diff
@@ -0,0 +1,60 @@
+ * An AddCommand consists of a Task.
+ *
+ */
+
```
The coding standard used for this module suggests leaving no blank line between the documentation block and the method/class. Perhaps it might save some space too?

deetoomoo commented on src/main/java/FindCommand.java:
```diff
@@ -0,0 +1,36 @@
+ *
+ */
+
+public class FindCommand extends Command{
```
There appears to be a missing whitespace here after Command :-)

deetoomoo commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,66 @@
+    public void complete(){
+        this.done = 1;
+    }
+
```
Maybe you should include javadocs for some of these public functions

deetoomoo reviewed (commented):
Overall good adherence to the java coding standard! Just some minor issues.. good job!

Haozeee commented on data/tasks.txt:
```diff
@@ -0,0 +1,2 @@
+D | 1 | return book  | Dec 02 2019, 06:00 PM
```
Perhaps this file should be included in the gitignore file since the tasks in it are only relevant to 1 particular user.

Haozeee commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,118 @@
+                return new AddCommand(new ToDo(task));
+            }
+        }
+        case "deadline": {
```
Personally, I think the case branch below is too long which makes the entire method looks unsightly and hard to read. Perhaps consider putting it into another method such as:
```java
private Command parseDeadline(String fullCommand)
```
This will make the code neater. Same thing for the Event case below.

Haozeee commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,101 @@
+
+    /**
+     * Reads and loads the storage into an ArrayList of Tasks.
+     * @return ArrayList of tasks stored in the storage.
```
There should be an empty line between the method and description and parameters according to the coding standard.

Haozeee commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,101 @@
+     * @throws FileNotFoundException if file path does not exist.
+     */
+
+    public ArrayList<Task> load() throws FileNotFoundException {
```
I think there is too much nesting in this method which makes the code hard to follow. Consider putting some of the if-else parts into seperate methods in order to simplify it.

Haozeee commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,66 @@
+ */
+
+public abstract class Task {
+    private int done;
```
Consider using a boolean instead to indicate whether a task is done or not.

Haozeee commented on src/main/java/ExitCommand.java:
```diff
@@ -0,0 +1,39 @@
+     * @see IOException
+     */
+
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws IOException {
```
There should be an override annotation here to indicate to the reader that this method is being overridden from a parent class. Same thing for the other classes that extend the Command abstract class.

Haozeee commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,61 @@
+     * @param index index of the task to be deleted.
+     */
+    public void delete(int index) {
+        tasks.remove(index);
```
I believe this will throw an exception that is not caught if the user specifies an index that is beyond the length of the ArrayList. May want to try it out by writing a test case for it.

Haozeee reviewed (commented):
Overall, I think the code is mostly easy to understand and follow. However, there are some things that can be improved. For instance, some methods can be made more concise to further improve code readability.  As for the coding standard, there should not be a space between a Javadoc comment and the method itself.

chengseong committed:
fixed issues with javaDocs and Coding Standards pointed out in pull request review

chengseong committed:
Added tasks.txt to .gitignore

chengseong committed:
Merge branch 'add-gradle-support' into branch-A-gradleSupport

chengseong committed:
Added gradle support

chengseong committed:
Fixed most checkstyle issues except NewlineAtEndofFile error

chengseong committed:
Fixed gradle issue and shifted Java files into packages

chengseong committed:
Created GUI

chengseong committed:
No assertions are being made in the code.

Assertions are important in identifying potential bugs in the code.

Let's add assertions to at certain points in the code.

Using assertions will allow us to ensure that none of the assumptions we
have about how the code should behave are being violated by the program
state.

chengseong committed:
Remove main method from Duke

chengseong committed:
Improved Code Quality

chengseong committed:
Merge pull request #3 from chengseong/branch-A-CodeQuality

branch-A-CodeQuality

chengseong committed:
Resolved merge conflicts

chengseong committed:
Resolved merge conflicts

chengseong committed:
Add Recurring Tasks for Event and Deadline Tasks

chengseong committed:
dummy commit

chengseong committed:
Added new Exceptions

chengseong committed:
Merge branch 'branch-A-Assertions'

chengseong committed:
Improved GUI

chengseong committed:
Added User Guide and User Interface Picture

chengseong committed:
Set theme jekyll-theme-slate

chengseong committed:
Set theme jekyll-theme-cayman

chengseong committed:
Set theme jekyll-theme-cayman

chengseong committed:
Update `bye` command to close window

chengseong committed:
Merge branch 'master' of https://github.com/chengseong/ip
